utils.py

```diff
@@ -0,0 +1,194 @@
+
+    for band_name in band_names:
+
+        band_data = hypercube[band_name]
+
+        mean_value = float(band_data.mean())
+        variance_value = float(band_data.var())
+        
+        min_value = float(band_data.min())
+        max_value = float(band_data.max())
+        
+        quantiles = band_data.quantile([0.25, 0.5, 0.75]).values
+        
+        statistics[band_name] = {
+            'mean': mean_value,
+            'variance': variance_value,
+            'min': min_value,
+            'max': max_value,
+            'quantile25': quantiles[0],
+            'quantile50': quantiles[1],
+            'quantile75': quantiles[2],
+        }
+
+    return statistics
+
+def assert_band_statistics(output_band_stats: dict, scenario_name: str, tolerance: float = 0.01) -> None:
+    """
+    Asserts the output band statistics against a reference dictionary stored in a JSON file with tolerance.
+
+    Parameters:
+        output_band_stats (dict): The output dictionary containing band statistics to be asserted.
+        reference_file (str): The path to the JSON file containing the reference dictionary.
+        tolerance (float): Tolerance value for comparing values. Defaults to 0.01.
+
+    Returns:
+        None
+    """
+    try:
+        groundtruth_band_stats = extract_reference_band_statistics(scenario_name)
+        compare_band_statistics(output_band_stats, groundtruth_band_stats, tolerance)
+    except FileNotFoundError:
+        print(f"Error: Reference file '{scenario_name}' not found.")
+    except json.JSONDecodeError:
+        print(f"Error: Unable to parse JSON in file '{scenario_name}'.")
+
+
+def execute_and_assert(cube: openeo.DataCube, 
+                       output_path: Union[str, Path], 
+                       scenario_name: str) -> None:
+    """
+    Avoid code duplication:
+    Execute the provided OpenEO cube, save the result to the output path, 
+    and assert its statistics against the reference data.
+
+    Parameters:
+        cube (openeo.datacube.DataCube): The OpenEO data cube to execute.
+        output_path (Union[str, Path]): The path where the output should be saved.
+        scenario_name (str): A name identifying the scenario for reference data.
+
+    Returns:
+        None
+
+    Raises:
+        RuntimeError: If there is an issue during execution, file saving, or assertion.
+    """
+
+    try:
+        cube.execute_batch(output_path,
+                           title=scenario_name,
+                           description='benchmarking-creo', 
+                           job_options={ 'driver-memory': '1g' }
+                           )
+
+        output_cube = xr.open_dataset(output_path)
+        output_band_stats = calculate_band_statistics(output_cube)
+
+        assert_band_statistics(output_band_stats, scenario_name)
+
+    except Exception as e:
+        raise RuntimeError(f"Error occurred during execution and assertion: {str(e)}")
```

Comment:
As mentioned above: don't do try-except constructs like this: let things fail directly . Rewriting exceptions makes post-mortem investigations harder